troubleshooting guide for 401 & 403 (#620)
kyma-project/api-gateway: docs/user/troubleshooting-guides/03-00-cannot-connect-to-service/03-01-401-unaythorized-403-forbidden.md

# 401 Unauthorized or 403 Forbidden

## Symptom

When you try to reach your service, you get `401 Unauthorized` or `403 Forbidden` in response.

## Cause 

The error `401 Unauthorized` occurs when you try to access a Service that requires authentication, but you have either not provided appropriate credentials or have not provided any credentials at all. You get the error `403 Forbidden` when you try to access a Service or perform an action for which you lack permission.

## Remedy

Make sure that you are using an access token with proper scopes, and it is active. Depending on the type of your access token, follow the relevant steps.

> ### JWT

Reviewer: ```suggestion
### JWT access token
```

> ### JWT

Reviewer: ```suggestion
### JSON Web Token
```

> 1. Decode JWT

Reviewer: ```suggestion
1. Decode the JWT access token.
```

> 1. Decode the JWT.

Reviewer: ```suggestion
1. Decode the JSON Web Token (JWT).
```

> 2. Check the validity and scopes of the access token:

Reviewer: ```suggestion
2. Check the validity and scopes of the JWT:
```

      ```bash
      {
         "sub": ********,
         "scp": "test",
         "aud": ********,
         "iss": ******,
         "exp": 1697120462,
         "iat": ******,
         "jti": ******,
      }
      ```

> 3. Generate a [new access token](../../tutorials/01-50-expose-and-secure-a-workload/01-51-get-jwt.md) if needed.

Reviewer: ```suggestion
3. Generate a [new JWT](../../tutorials/01-50-expose-and-secure-a-workload/01-51-get-jwt.md) if needed.
```
